### PR TITLE
Add MarkdownBear and format existing .md

### PR DESCRIPTION
**Pull request by blazeu**
Default MarkdownBear style used.
Probably need CI.

Closes https://github.com/coala/cEPs/issues/125

### CODE REVIEW

File: cEP-0000.md [hunk 1]
```diff
@@ -1,17 +1,15 @@
...
-|Title   |coala Enhancement Proposals        |
-|Authors |Lasse Schuirmann <lasse@gitmate.io>|
-|Status  |Active                             |
-|Type    |Process                            |
+| Metadata |                                            |
+| -------- | ------------------------------------------ |
+| cEP      | 0000                                       |
+| Version  | 2.2                                        |
+| Title    | coala Enhancement Proposals                |
+| Authors  | Lasse Schuirmann <mailto:lasse@gitmate.io> |
```
Comment (blazeu) [inline cEP-0000.md:8]: Not sure if we want this, since GitHub already highlight email without `mailto:`
but MarkdownBear enforces this

**Comment by blazeu** [comment]
> Note that MarkdownBear often gives us trouble, so this may not be as easy as it sounds.
https://github.com/coala/cEPs/issues/125#issuecomment-386891802

There's this bug https://github.com/coala/coala-bears/issues/1235 which can *usually* be avoided by installing the correct packages and version of remark.

Also, https://github.com/coala/coala-bears/issues/2131#issuecomment-387429654 which can be confusing as hell, upstream bug and needs more testing.

File: cEP-0014.md [hunk 2]
```diff
@@ -341,25 +463,38 @@ A possible set of deault capabilties can be:
...
+   which, the bear may be discarded, or any one of the possible value is filled
+   in with a warning comment.
 6. The per-language sections are generated, and final `.coafile` is written.
 
-
 ## Code Samples
 
 ### Collecting extracted information from project
+
+<!-- Start ignoring MarkdownBear -->
```
Comment (blazeu) [inline cEP-0014.md:497]: Maybe a bug, it wants to add more and more indentation to the codeblocks below.

File: cEP-0014.md [hunk 3]
```diff
@@ -410,7 +545,8 @@ def aggregate_info(infoextractors):
 ```
 
 ### Collecting all the meta-data
-```py
+
+````py
```
Comment (blazeu) [inline cEP-0014.md]: 4 backticks? I don't understand the reason.

File: cEP-0010.md [hunk 4]
```diff
@@ -122,155 +120,161 @@ in snake_case).
...
+  used in commit message shortlog.
+- `Root.Metadata.CommitMessage.Shortlog.Length`: This aspect checks on the
+  length of your commit message shortlog given the value of
+  `Root.Metadata.CommitMessage.Shortlog.Length.max_shortlog_length`.
+- `Root.Metadata.CommitMessage.Shortlog.FirstCharacter`: This aspect detects
+  inconsistent casing for the first character of your commit message
+  shortlog(and the the first character following the colon if there is one)
+  given the value of
+  <!-- Ignore MarkdownBear -->
+  `Root.Metadata.CommitMessage.Shortlog.FirstCharacter.shortlog_starts_upper_case`
```
Comment (blazeu) [inline cEP-0010.md:248]: MarkdownBear wants to warp inline codeblock, this doesn't happen with `remark-lint@6`, but MarkdownBear is using v5.

File: cEP-0000.md (hunk 1 above)
Comment (jayvdb) [reply cEP-0000.md:8]: Seems to be useful; see https://github.com/remarkjs/remark-lint/tree/master/packages/remark-lint-no-auto-link-without-protocol

File: cEP-0001.md [hunk 5]
```diff
@@ -1,25 +1,20 @@
...
-|Authors |Maximilian Scholz <max.scholz@mailbox.org>, Lasse Schuirmann <lasse@gitmate.io>, Mischa Krüger <makman@alice.de>|
-|Status  |Active                                                              |
-|Type    |Process                                                             |
+| Metadata |                                                                                                                                       |
+| -------- | ------------------------------------------------------------------------------------------------------------------------------------- |
+| cEP      | 0001                                                                                                                                  |
+| Version  | 2.0                                                                                                                                   |
+| Title    | Decision Making in coala                                                                                                              |
+| Authors  | Maximilian Scholz <mailto:max.scholz@mailbox.org>, Lasse Schuirmann <mailto:lasse@gitmate.io>, Mischa Krüger <mailto:makman@alice.de> |
```
Comment (jayvdb) [inline cEP-0001.md:8]: this is going to be annoying when another author is added.
markdown tables suck

File: cEP-0001.md [hunk 6]
```diff
@@ -56,12 +51,12 @@ A Maintainer is someone that showed persistence in contributions and or an
...
 
-A developer can be promoted to a maintainer with a B level decision. This decision
-does not have to be initiated by a maintainer so it is valid for a developer to
-propose himself for a promotion.
-A maintainer has all the rights a developer has. Additionally maintainers have
-write access to coala repositories, can promote newcomers and developers and
-have voting rights on B and C level decisions.
+A developer can be promoted to a maintainer with a B level decision. This
+decision does not have to be initiated by a maintainer so it is valid for a
+developer to propose himself for a promotion. A maintainer has all the rights a
```
Comment (jayvdb) [inline cEP-0001.md]: add an extra blank line before `A maintainer ..` - that preserves the intent a bit better IMO

File: cEP-0001.md (hunk 5 above)
Comment (blazeu) [reply cEP-0001.md:8]: Make it 1 author per line?

File: cEP-0014.md (hunk 3 above)
Comment (jayvdb) [reply cEP-0014.md]: Agree  strange.
You intended to put an `Start ignoring..` here?

**Review by jayvdb — CHANGES_REQUESTED** [review]

File: cEP-0001.md (hunk 5 above)
Comment (jayvdb) [reply cEP-0001.md:8]: then we get into variations of markdown.
not worth it atm.
IMO we propose a change to cEP 0 to replace .md with .asciidoc or rst. 

https://github.com/coala/cEPs/issues/129

File: cEP-0014.md (hunk 3 above)
Comment (blazeu) [reply cEP-0014.md]: Okay, replaced it with 3 and MarkdownBear didn't complain

**Review by jayvdb — DISMISSED** [review]

**Comment by jayvdb** [comment]
ack https://github.com/coala/cEPs/pull/128/commits/3ab92cac93f1147f0be1615721b318a0c5038fed

**Comment by jayvdb** [comment]
@gitmate-bot rebase

**Review by jayvdb — APPROVED** [review]

**Comment by jayvdb** [comment]
reack https://github.com/coala/cEPs/pull/128/commits/c1a4b7e48a5d29503d79e89f08b6e1a9ee26495b

**Comment by jayvdb** [comment]
@gitmate-bot ff

**Comment by virresh** [comment]
I tried running the mark-down bear locally, but it reports errors in other cEP's as well
Here is the corresponding output https://gist.github.com/virresh/b65304112e75a238f00a41a9f3d00a86

I used `remark-lint@5.0.0` and `remark-cli@2.0.0`

**Comment by blazeu** [comment]
@virresh can you open an issue please?

**Comment by virresh** [comment]
@blazeu I realised I was using coala's development version
Using the pip version solved it :sweat_smile: 

Although even the pre-release on pip doesn't work well with MarkDown bear
Probably we will have to look into it when we do the next pre-release of coala :sweat_smile:

**Pull request merged**